File: ruby/problem_4.rb
answer = 0
for x in 900 .. 1000 do
    for y in 990 .. 1000 do
        tmp = x * y
        stmp = String(tmp)
        if stmp == stmp.reverse
            answer = tmp
        end
    end
end

puts("Answer 4: " + String(answer))

File: python/problem_4.py
from __future__ import print_function

answer = 0
for x in range(900, 1000):
    for y in range(990, 1000):
        tmp = x * y
        stmp = str(tmp)
        if stmp == stmp[::-1]:
            answer = tmp

print("Answer 4:", answer)

